test/codegen/query_thread_pool_test.cpp

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// query_thread_pool_test.cpp
+//
+// Identification: test/codegen/query_thread_pool_test.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/codegen.h"
+#include "codegen/codegen_test_util.h"
+#include "codegen/multi_thread_context.h"
+#include "codegen/query_thread_pool.h"
+#include "common/harness.h"
+
+namespace peloton {
+namespace test {
+
+//===--------------------------------------------------------------------===//
+// Query Thread Pool Test
+//===--------------------------------------------------------------------===//
+
+class QueryThreadPoolTests : public PelotonTest {};
+
+TEST_F(QueryThreadPoolTests, BasicTest) {
+  codegen::QueryThreadPool thread_pool(4);
+
+  codegen::CodeContext code_context;
+  codegen::CodeGen codegen(code_context);
+
+  std::atomic<bool> finished(false);
+  codegen::MultiThreadContext context(codegen.Const64(0), codegen.Const64(4), &finished);
+
+  thread_pool.SubmitQueryTask(context, [](__attribute__ ((unused)) codegen::MultiThreadContext context){
```

Comment:
Remove the parameter as @phisiart said. Fix in 4ceba838d251c5b893cbee7cd1595af10689e159, thx.